lib/event_store_client/connection.rb

```diff
@@ -79,5 +73,47 @@ def client
           host: host, port: port, per_page: per_page
         )
     end
+
+    def read_from_stream(stream, direction:, start:, count:, resolve_links:)
+      response =
+        client.read(
+          stream, start: start, direction: direction, count: count, resolve_links: resolve_links
+        )
+      return [] if response.body.nil? || response.body.empty?
+      JSON.parse(response.body)['entries'].map do |entry|
+        deserialize_event(entry)
+      end.reverse
+    end
+
+    def read_all_from_stream(stream, resolve_links:)
+      count = 20
+      start = 0
+      events = []
+
+      loop do
+        response =
+          client.read(
+            stream, start: start, direction: 'forward', count: count, resolve_links: resolve_links
+          )
+        break if response.body.nil? || response.body.empty?
+        entries = JSON.parse(response.body)['entries']
+        break if entries.empty?
+        deserialzied_events = entries.map { |entry| deserialize_event(entry) }
+        events = events + deserialzied_events.reverse
```

Comment:
I would probably did like `events += entries.map { |entry| deserialize_event(entry) }.reverse`  so would to avoid creating inremediate variable `deserialzied_events `

still, need reverse?

Reply:
Event Store returns events in order from the latest to the oldest. You have to reverse the returned collection in order to return a proper collection.  It's forward reading.
For the count set to 2 the event store returns events with id in the way as below:
first get: [1, 0]
second get: [3, 2]
Without the `revers`, your final collection would look like this [1, 0, 3, 2]
With `revers`: [0, 1, 2, 3]